patches/server/0839-Add-paper-bugreport-command.patch

```diff
@@ -0,0 +1,106 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: simpleauthority <jacob@algorithmjunkie.com>
+Date: Fri, 24 Dec 2021 14:04:23 -0800
+Subject: [PATCH] Add "/paper bugreport" command.
+
+This fulfills the feature request in issue #5347 and prefills datapacks, plugins, and the server version into the Paper bug report template.
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperCommand.java b/src/main/java/com/destroystokyo/paper/PaperCommand.java
+index c0d123bff1825366c30aadd3ad8a7fde68ef74e4..2c967c4a92be1d0ca5a5a061eb1a3763db167ca4 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperCommand.java
++++ b/src/main/java/com/destroystokyo/paper/PaperCommand.java
+@@ -11,12 +11,15 @@ import com.google.common.collect.Maps;
+ import com.google.gson.JsonObject;
+ import com.google.gson.internal.Streams;
+ import com.google.gson.stream.JsonWriter;
++import io.papermc.paper.datapack.Datapack;
+ import net.kyori.adventure.text.Component;
+ import net.kyori.adventure.text.ComponentLike;
+ import net.kyori.adventure.text.JoinConfiguration;
+ import net.kyori.adventure.text.TextComponent;
++import net.kyori.adventure.text.event.ClickEvent;
+ import net.kyori.adventure.text.format.NamedTextColor;
+ import net.kyori.adventure.text.format.TextColor;
++import net.kyori.adventure.text.serializer.plain.PlainTextComponentSerializer;
+ import net.minecraft.core.Registry;
+ import net.minecraft.resources.ResourceLocation;
+ import net.minecraft.server.MCUtil;
+@@ -47,11 +50,14 @@ import org.bukkit.craftbukkit.entity.CraftPlayer;
+ import org.bukkit.craftbukkit.inventory.CraftItemStack;
+ import org.bukkit.entity.Player;
+ import org.bukkit.inventory.ItemStack;
++import org.bukkit.plugin.Plugin;
+ 
+ import java.io.File;
+ import java.io.FileOutputStream;
+ import java.io.PrintStream;
+ import java.io.StringWriter;
++import java.net.URLEncoder;
++import java.nio.charset.StandardCharsets;
```

Comment:
And yea, regarding these I prefer to just always use the fully qualified names, as imports cause a lot of noise in ur patch.
But, I mean if you use it many times you can always import it if you want. But usually, it hardly makes a difference...

Reply:
Fair point. It's just some of the FQNs seem pretty long and it looks gross to me but in the spirit of achieving minimal diff I guess some sacrifices must be made. :P